client/my-sites/upgrades/checkout/credit-card-payment-box.jsx

```diff
@@ -12,17 +13,69 @@ var PayButton = require( './pay-button' ),
 	TermsOfService = require( './terms-of-service' ),
 	PaymentBox = require( './payment-box' ),
 	analytics = require( 'lib/analytics' ),
-	cartValues = require( 'lib/cart-values' );
+	cartValues = require( 'lib/cart-values' ),
+	transactionStepTypes = require( 'lib/store-transactions/step-types' );
 
 import CartCoupon from 'my-sites/upgrades/cart/cart-coupon';
 import PaymentChatButton from './payment-chat-button';
 import config from 'config';
 import { PLAN_BUSINESS } from 'lib/plans/constants';
-import { some } from 'lodash';
+import ProgressBar from 'components/progress-bar';
 
 var CreditCardPaymentBox = React.createClass( {
 	getInitialState: function() {
-		return { previousCart: null };
+		return {
+			elapsed: 0,
+			previousCart: null
+		};
+	},
+
+	componentWillReceiveProps: function( nextProps ) {
+		if ( nextProps.transactionStep && this.submitting( nextProps.transactionStep ) ) {
+			this.timer = setInterval( this.tick, 100 );
+		}
+	},
+
+	componentWillUnmount: function() {
+		clearInterval( this.timer );
+	},
+
+	tick: function() {
+		const elasped = this.state.elapsed + 1 / 200 * ( 100 - this.state.elapsed );
```

Comment:
I think a comment like `// increase the progress bar by 0.5% of the remaining progress` or something might be helpful here for future readers.